<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CTCF introduction • CTCF</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="CTCF introduction">
<meta property="og:description" content="CTCF">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CTCF</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.00.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/CTCF.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dozmorovlab/CTCF/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>CTCF introduction</h1>
                        <h4 data-toc-skip class="author">Mikhail
Dozmorov</h4>
            <address class="author_afil">
      Virginia Commonwealth
University<br><a class="author_email" href="mailto:#"></a><a href="mailto:mikhail.dozmorov@gmail.com" class="email">mikhail.dozmorov@gmail.com</a>
      </address>
                  
            <h4 data-toc-skip class="date">6 November 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dozmorovlab/CTCF/blob/HEAD/vignettes/CTCF.Rmd" class="external-link"><code>vignettes/CTCF.Rmd</code></a></small>
      <div class="hidden name"><code>CTCF.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="ctcf">CTCF<a class="anchor" aria-label="anchor" href="#ctcf"></a>
</h2>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable" class="external-link"><img src="https://img.shields.io/badge/lifecycle-stable-brightgreen.svg" alt="Lifecycle: stable"></a>
<!-- [![BioC status](http://www.bioconductor.org/shields/build/release/bioc/CTCF.svg)](https://bioconductor.org/checkResults/release/bioc-LATEST/CTCF) [![R-CMD-check-bioc](https://github.com/mdozmorov/CTCF/actions/workflows/R-CMD-check-bioc.yaml/badge.svg)](https://github.com/mdozmorov/CTCF/actions/workflows/R-CMD-check-bioc.yaml) --></p>
<p><em><a href="https://bioconductor.org/packages/3.16/CTCF" class="external-link">CTCF</a></em> defines
an AnnotationHub resource representing genomic coordinates of <a href="https://meme-suite.org/meme/doc/fimo.html" class="external-link">FIMO</a>-predicted CTCF
binding sites for human and mouse genomes, including the <a href="https://github.com/marbl/CHM13" class="external-link">Telomere-to-Telomere</a> and <a href="http://genomeref.blogspot.com/2020/07/grcm39-new-mouse-reference-genome.html" class="external-link">mm39</a>
genome assemblies. It also includes experimentally defined CTCF-bound
cis-regulatory elements from <a href="https://screen.encodeproject.org/" class="external-link">ENCODE SCREEN</a>.</p>
<p>TL;DR - for human hg38 genome assembly, use
<code>hg38.MA0139.1.RData</code> (“AH104729”). For mouse mm10 genome
assembly, use <code>mm10.MA0139.1.RData</code> (“AH104755”). For <a href="https://screen.encodeproject.org/" class="external-link">ENCODE SCREEN</a> data, use
<code>hg38.SCREEN.GRCh38_CTCF.RData</code> (“AH104730”) or
<code>mm10.SCREEN.mm10_CTCF.RData</code> (“AH104756”) objects.</p>
<p>The CTCF GRanges are named as
<code>&lt;assembly&gt;.&lt;Database&gt;</code>. The FIMO-predicted data
includes extra columns with motif name, score, p-value, q-value, and the
motif sequence.</p>
<!--
**Please, note that the updated CTCF objects will be available in 
Bioconductor/AnnotationHub 3.16.** To test the following code, use the 
`bioconductor::devel` Docker image. Run:

```bash
docker run -e PASSWORD=password -p 8787:8787 -d --rm -v $(pwd):/home/rstudio bioconductor/bioconductor_docker:devel
```
Open http://localhost:8787 and login using `rstudio/password` credentials.
-->
<div class="section level3">
<h3 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h3>
<p><a href="https://www.bioconductor.org/install/#install-R" class="external-link">Install</a>
the latest release of R, then get the latest version of Bioconductor by
starting R and entering the commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if (!require("BiocManager", quietly = TRUE))</span></span>
<span><span class="co">#     install.packages("BiocManager")</span></span>
<span><span class="co"># BiocManager::install(version = "3.16")</span></span></code></pre></div>
<p>Then, install additional packages using the following code:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># BiocManager::install("AnnotationHub", update = FALSE) </span></span>
<span><span class="co"># BiocManager::install("GenomicRanges", update = FALSE)</span></span>
<span><span class="co"># BiocManager::install("plyranges", update = FALSE)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; snapshotDate(): 2022-10-26</span></span>
<span><span class="va">query_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">ah</span>, <span class="va">preparerclass</span> <span class="op">==</span> <span class="st">"CTCF"</span><span class="op">)</span></span>
<span><span class="co"># Explore the AnnotationHub object</span></span>
<span><span class="va">query_data</span></span>
<span><span class="co">#&gt; AnnotationHub with 51 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-10-26</span></span>
<span><span class="co">#&gt; # $dataprovider: JASPAR 2022, CTCFBSDB 2.0, SwissRegulon, Jolma 2013, HOCOMO...</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens, Mus musculus</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH104716"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                                 </span></span>
<span><span class="co">#&gt;   AH104716 | T2T.CIS_BP_2.00_Homo_sapiens.RData                    </span></span>
<span><span class="co">#&gt;   AH104717 | T2T.CTCFBSDB_PWM.RData                                </span></span>
<span><span class="co">#&gt;   AH104718 | T2T.HOCOMOCOv11_core_HUMAN_mono_meme_format.RData     </span></span>
<span><span class="co">#&gt;   AH104719 | T2T.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</span></span>
<span><span class="co">#&gt;   AH104720 | T2T.Jolma2013.RData                                   </span></span>
<span><span class="co">#&gt;   ...        ...                                                   </span></span>
<span><span class="co">#&gt;   AH104762 | mm9.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</span></span>
<span><span class="co">#&gt;   AH104763 | mm9.Jolma2013.RData                                   </span></span>
<span><span class="co">#&gt;   AH104764 | mm9.MA0139.1.RData                                    </span></span>
<span><span class="co">#&gt;   AH104765 | mm9.SwissRegulon_human_and_mouse.RData                </span></span>
<span><span class="co">#&gt;   AH104766 | mm8.CTCFBSDB.CTCF_predicted_mouse.RData</span></span>
<span><span class="co"># Get the list of data providers</span></span>
<span><span class="va">query_data</span><span class="op">$</span><span class="va">dataprovider</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "CIS-BP"           "CTCFBSDB 2.0"     "HOCOMOCO v11"     "JASPAR 2022"     </span></span>
<span><span class="co">#&gt; [5] "Jolma 2013"       "SwissRegulon"     "ENCODE SCREEN v3"</span></span></code></pre></div>
<p>We can find CTCF sites identified using JASPAR 2022 database in hg38
human genome</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">query_data</span>, <span class="va">species</span> <span class="op">==</span> <span class="st">"Homo sapiens"</span> <span class="op">&amp;</span> </span>
<span>                   <span class="va">genome</span> <span class="op">==</span> <span class="st">"hg38"</span> <span class="op">&amp;</span> </span>
<span>                   <span class="va">dataprovider</span> <span class="op">==</span> <span class="st">"JASPAR 2022"</span><span class="op">)</span></span>
<span><span class="co">#&gt; AnnotationHub with 2 records</span></span>
<span><span class="co">#&gt; # snapshotDate(): 2022-10-26</span></span>
<span><span class="co">#&gt; # $dataprovider: JASPAR 2022</span></span>
<span><span class="co">#&gt; # $species: Homo sapiens</span></span>
<span><span class="co">#&gt; # $rdataclass: GRanges</span></span>
<span><span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span></span>
<span><span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span></span>
<span><span class="co">#&gt; #   rdatapath, sourceurl, sourcetype </span></span>
<span><span class="co">#&gt; # retrieve records with, e.g., 'object[["AH104727"]]' </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;              title                                                  </span></span>
<span><span class="co">#&gt;   AH104727 | hg38.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</span></span>
<span><span class="co">#&gt;   AH104729 | hg38.MA0139.1.RData</span></span>
<span><span class="co"># Same for mm10 mouse genome</span></span>
<span><span class="co"># subset(query_data, species == "Mus musculus" &amp; genome == "mm10" &amp; dataprovider == "JASPAR 2022")</span></span></code></pre></div>
<p>The <code>hg38.JASPAR2022_CORE_vertebrates_non_redundant_v2</code>
object contains CTCF sites detected using the <a href="https://jaspar.genereg.net/search?q=CTCF&amp;collection=all&amp;tax_group=all&amp;tax_id=9606&amp;type=all&amp;class=all&amp;family=all&amp;version=all" class="external-link">all
three CTCF PWMs</a>. To retrieve, we’ll use:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># hg38.JASPAR2022_CORE_vertebrates_non_redundant_v2</span></span>
<span><span class="va">CTCF_hg38_all</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH104727"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="co">#&gt; require("GenomicRanges")</span></span>
<span><span class="va">CTCF_hg38_all</span></span>
<span><span class="co">#&gt; GRanges object with 3093041 ranges and 5 metadata columns:</span></span>
<span><span class="co">#&gt;             seqnames            ranges strand |                   name</span></span>
<span><span class="co">#&gt;                &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt;</span></span>
<span><span class="co">#&gt;         [1]     chr1       11212-11246      + | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;         [2]     chr1       11399-11432      + | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;         [3]     chr1       11414-11432      + | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;         [4]     chr1       12373-12406      + | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;         [5]     chr1       13507-13541      + | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;         ...      ...               ...    ... .                    ...</span></span>
<span><span class="co">#&gt;   [3093037]     chrY 57215115-57215148      - | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;   [3093038]     chrY 57215146-57215164      - | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;   [3093039]     chrY 57215146-57215179      - | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;   [3093040]     chrY 57215332-57215366      - | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;   [3093041]     chrY 57216319-57216352      - | JASPAR2022_CORE_vert..</span></span>
<span><span class="co">#&gt;                 score    pvalue    qvalue               sequence</span></span>
<span><span class="co">#&gt;             &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;            &lt;character&gt;</span></span>
<span><span class="co">#&gt;         [1]   7.77064  5.25e-05     0.459 gtgctgtgccagggcgcccc..</span></span>
<span><span class="co">#&gt;         [2]  18.48780  2.54e-07     0.118 cagcacgcccacctgctggc..</span></span>
<span><span class="co">#&gt;         [3]   9.11475  5.65e-05     0.555    ctggcagctggggacactg</span></span>
<span><span class="co">#&gt;         [4]   9.21951  5.25e-05     0.421 CAGCAGGTCTGGCTTTGGCC..</span></span>
<span><span class="co">#&gt;         [5]   9.71560  2.24e-05     0.397 GTGCCCTTCCTTTGCTCTGC..</span></span>
<span><span class="co">#&gt;         ...       ...       ...       ...                    ...</span></span>
<span><span class="co">#&gt;   [3093037]   8.06504  9.11e-05     0.614 CTGCTGGGCCCTCTTGCTCC..</span></span>
<span><span class="co">#&gt;   [3093038]   9.11475  5.65e-05     0.726    CTGGCAGCTGGGGACACTG</span></span>
<span><span class="co">#&gt;   [3093039]  17.91870  3.72e-07     0.246 CAGCACGCCCGCCTGCTGGC..</span></span>
<span><span class="co">#&gt;   [3093040]   7.77064  5.25e-05     0.584 GTGCTGTGCCAGGGCGCCCC..</span></span>
<span><span class="co">#&gt;   [3093041]   8.63415  6.96e-05     0.595 CTGCATTTGCGTTCCGACGC..</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from hg38 genome</span></span></code></pre></div>
<p>The <code>hg38.MA0139.1</code> object contains CTCF sites detected
using the most popular <a href="https://jaspar.genereg.net/matrix/MA0139.1/" class="external-link">MA0139.1</a> CTCF
PWM. To retrieve:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># hg38.MA0139.1</span></span>
<span><span class="va">CTCF_hg38</span> <span class="op">&lt;-</span> <span class="va">query_data</span><span class="op">[[</span><span class="st">"AH104729"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; loading from cache</span></span>
<span><span class="va">CTCF_hg38</span></span>
<span><span class="co">#&gt; GRanges object with 887980 ranges and 5 metadata columns:</span></span>
<span><span class="co">#&gt;            seqnames            ranges strand |        name     score    pvalue</span></span>
<span><span class="co">#&gt;               &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt;        [1]     chr1       11414-11432      + |    MA0139.1   9.11475  5.65e-05</span></span>
<span><span class="co">#&gt;        [2]     chr1       14316-14334      + |    MA0139.1   7.83607  9.71e-05</span></span>
<span><span class="co">#&gt;        [3]     chr1       15439-15457      + |    MA0139.1   8.00000  9.08e-05</span></span>
<span><span class="co">#&gt;        [4]     chr1       16603-16621      + |    MA0139.1   8.04918  8.89e-05</span></span>
<span><span class="co">#&gt;        [5]     chr1       16651-16669      + |    MA0139.1  11.42620  1.97e-05</span></span>
<span><span class="co">#&gt;        ...      ...               ...    ... .         ...       ...       ...</span></span>
<span><span class="co">#&gt;   [887976]     chrY 57209918-57209936      - |    MA0139.1  11.42620  1.97e-05</span></span>
<span><span class="co">#&gt;   [887977]     chrY 57209966-57209984      - |    MA0139.1   8.04918  8.89e-05</span></span>
<span><span class="co">#&gt;   [887978]     chrY 57211133-57211151      - |    MA0139.1   8.00000  9.08e-05</span></span>
<span><span class="co">#&gt;   [887979]     chrY 57212256-57212274      - |    MA0139.1   7.83607  9.71e-05</span></span>
<span><span class="co">#&gt;   [887980]     chrY 57215146-57215164      - |    MA0139.1   9.11475  5.65e-05</span></span>
<span><span class="co">#&gt;               qvalue            sequence</span></span>
<span><span class="co">#&gt;            &lt;numeric&gt;         &lt;character&gt;</span></span>
<span><span class="co">#&gt;        [1]     0.555 ctggcagctggggacactg</span></span>
<span><span class="co">#&gt;        [2]     0.601 GGACCAACAGGGGCAGGAG</span></span>
<span><span class="co">#&gt;        [3]     0.599 TAGCCTCCAGAGGCCTCAG</span></span>
<span><span class="co">#&gt;        [4]     0.597 CCACCTGAAGGAGACGCGC</span></span>
<span><span class="co">#&gt;        [5]     0.504 TGGCCTACAGGGGCCGCGG</span></span>
<span><span class="co">#&gt;        ...       ...                 ...</span></span>
<span><span class="co">#&gt;   [887976]     0.648 TGGCCTACAGGGGCCGCGG</span></span>
<span><span class="co">#&gt;   [887977]     0.770 CCACCTGAAGGAGACGCGC</span></span>
<span><span class="co">#&gt;   [887978]     0.770 TAGCCTCCAGAGGCCTCAG</span></span>
<span><span class="co">#&gt;   [887979]     0.770 GGACCAACAGGGGCAGGAG</span></span>
<span><span class="co">#&gt;   [887980]     0.726 CTGGCAGCTGGGGACACTG</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 24 sequences from hg38 genome</span></span></code></pre></div>
<p>It is always advisable to sort GRanges objects and keep standard
chromsomes:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">plyranges</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">CTCF_hg38_all</span> <span class="op">&lt;-</span> <span class="va">CTCF_hg38_all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">CTCF_hg38</span> <span class="op">&lt;-</span> <span class="va">CTCF_hg38</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">keepStandardChromosomes</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Save the data in a BED file, if needed.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note that rtracklayer::import and rtracklayer::export perform unexplained</span></span>
<span><span class="co"># start coordinate conversion, likely related to 0- and 1-based coordinate</span></span>
<span><span class="co"># system. We recommend converting GRanges to a data frame and save tab-separated</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">CTCF_hg38</span><span class="op">)</span>, </span>
<span>            file <span class="op">=</span> <span class="st">"CTCF_hg38.bed"</span>,</span>
<span>            sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>Note that the <a href="https://meme-suite.org/meme/doc/fimo.html" class="external-link">FIMO</a> tool detects
CTCF binding sites using the 1e-4 p-value threshold by default (the more
significant p-value corresponds to the more confidently detected CTCF
motif). We found that this threshold may be too permissive. Using the <a href="https://screen.encodeproject.org/" class="external-link">ENCODE SCREEN</a> database as
ground truth, we found 1e-6 as the optimal threshold providing
approximately 80% true positive rate. However, less significant CTCF
motifs may be cell type-specific or have weaker CTCF binding and
therefore be missed by conventional peak callers. If cell type-specific
CTCF binding is of interest, we recommend exploring less significant
CTCF sites.</p>
<p><img src="../reference/figures/Figure_human_pvalues_threshold.png" width="652"></p>
<p>To filter the GRanges object and keep high-confidence CTCF sites,
use:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Check length before filtering</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of CTCF motifs at the default 1e-4 threshold:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CTCF_hg38</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of CTCF motifs at the default 1e-4 threshold: 887980"</span></span>
<span><span class="co"># Filter and check length after filtering</span></span>
<span><span class="va">CTCF_hg38_filtered</span> <span class="op">&lt;-</span> <span class="va">CTCF_hg38</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">1e-6</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of CTCF motifs at the 1e-6 threshold:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CTCF_hg38_filtered</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of CTCF motifs at the 1e-6 threshold: 21671"</span></span>
<span><span class="co"># Similarly, filter</span></span>
<span><span class="va">CTCF_hg38_all_filtered</span> <span class="op">&lt;-</span> <span class="va">CTCF_hg38_all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">plyranges</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">pvalue</span> <span class="op">&lt;</span> <span class="fl">1e-6</span><span class="op">)</span></span></code></pre></div>
<p>Given some databases provide multiple CTCF PWMs, one CTCF site may be
detected multiple times resulting in overlapping CTCF sites. For
example, the proportion of overlapping CTCF sites in the
<code>CTCF_hg38_all_filtered</code> object containing CTCF sites
detected by three matrices nearly 40%:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Proportion of overlapping enrtries</span></span>
<span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">CTCF_hg38_all</span>, <span class="va">CTCF_hg38_all</span><span class="op">)</span></span>
<span><span class="va">prop_overlap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">queryHits</span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">prop_overlap</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prop_overlap</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"1"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CTCF_hg38_all</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.375093</span></span></code></pre></div>
<p>The proportion of overlapping CTCF sites in the
<code>CTCF_hg38_filtered</code> object containing CTCF sites detected by
the <a href="https://jaspar.genereg.net/matrix/MA0139.1/" class="external-link">MA0139.1</a>
matrix is less than 2.5%</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tmp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span><span class="va">CTCF_hg38</span>, <span class="va">CTCF_hg38</span><span class="op">)</span></span>
<span><span class="va">prop_overlap</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu">queryHits</span><span class="op">(</span><span class="va">tmp</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">prop_overlap</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">prop_overlap</span><span class="op">)</span> <span class="op">!=</span> <span class="st">"1"</span><span class="op">)</span><span class="op">]</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CTCF_hg38</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.0233485</span></span></code></pre></div>
<p>Reducing them (merging overlapping CTCF sites), combined with 1E-6
cutoff filtering, yields the number of CTCF sites comparable to
previously reported.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of CTCF_hg38 motifs at the 1e-6 threshold AND reduced:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CTCF_hg38_filtered</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of CTCF_hg38 motifs at the 1e-6 threshold AND reduced: 21652"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Number of CTCF_hg38_all motifs at the 1e-6 threshold AND reduced:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">CTCF_hg38_all_filtered</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/inter-range-methods.html" class="external-link">reduce</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Number of CTCF_hg38_all motifs at the 1e-6 threshold AND reduced: 63572"</span></span></code></pre></div>
<p>However, regulatory elements with CTCF proteins co-occupying
adjacent/overlapping CTCF binding motifs were shown to be functionally
and structurally different from those with single CTCF motifs. We
provide non-reduced CTCF data and advise considering overlap of CTCF
sites depending on the study’s goal.</p>
</div>
</div>
<div class="section level2">
<h2 id="liftover-of-ctcf-coordinates">liftOver of CTCF coordinates<a class="anchor" aria-label="anchor" href="#liftover-of-ctcf-coordinates"></a>
</h2>
<p>As genome assemblies for model organisms continue to improve, CTCF
sites for previous genome assemblies become obsolete. Typically, the
actual genome sequence changes little, leading to changes in genomic
coordinates. The <a href="https://genome.ucsc.edu/cgi-bin/hgLiftOver" class="external-link">liftOver</a> method
allows for conversion of genomic coordinates between genome
assemblies.</p>
<p>Some carefully curated CTCF sites are available only for older genome
assemblies. Examples include the data from <a href="https://insulatordb.uthsc.edu/" class="external-link">CTCFBSDB</a>, available for hg18
and mm8 genome assemblies.</p>
<p>To investigate whether liftOver of CTCF sites from older genome
assemblies is a viable option, we tested for overlap between CTCF sites
directly detected in specific genome assemblies with those lifted over.
We detected CTCF sites using the <a href="https://jaspar.genereg.net/matrix/MA0139.1/" class="external-link">MA0139.1</a> PWM from
JASPAR 2022 database in hg18, hg19, hg38, and T2T genome assemblies and
converted their genomic coordinates using the corresponding liftOver
chains (<a href="https://github.com/dozmorovlab/CTCF.dev/blob/main/scripts/download_liftOver.sh" class="external-link">download_liftOver.sh</a>
and <a href="https://github.com/dozmorovlab/CTCF.dev/blob/main/scripts/convert_liftOver.sh" class="external-link">convert_liftOver.sh</a>
scripts). We observed high Jaccard overlap among CTCF sites detected in
the original genome assemblies or lifted over.</p>
<p><strong>Jaccard overlaps among CTCF binding sites detected in the
original and liftOver human genome assemblies.</strong> CTCF sites were
detected using JASPAR 2022 MA0139.1 PWM. The correlogram was clustered
using Euclidean distance and Ward.D clustering . White-red gradient
indicate low-to-high Jaccard overlaps. Jaccard values are shown in the
corresponding cells.</p>
<p><img src="../reference/figures/Figure_liftOverJaccard.png" width="1125"></p>
<p>Our results suggest that liftOver is a viable alternative to obtain
CTCF genomic annotations for different genome assemblies. We provide <a href="https://insulatordb.uthsc.edu/" class="external-link">CTCFBSDB</a> data converted to
hg19 and hg38 genome assemblies.</p>
</div>
<div class="section level2">
<h2 id="ctcf-position-weight-matrices">CTCF Position Weight Matrices<a class="anchor" aria-label="anchor" href="#ctcf-position-weight-matrices"></a>
</h2>
<p><strong>CTCF PWM information.</strong> “Motif” - individual motif IDs
or the total number of motifs per database; “Length” - motif length of
the range of lengths; “URL” - direct links to motif pages. Jaspar,
Hocomoco, Jolma 2013 PWMs were downloaded from the <a href="https://meme-suite.org/meme/doc/download.html" class="external-link">MEME
database</a>.</p>
<table class="table">
<colgroup>
<col width="36%">
<col width="8%">
<col width="55%">
</colgroup>
<thead><tr class="header">
<th align="left">Motif</th>
<th align="left">Length..bp.</th>
<th align="left">URL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Jaspar2022</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">MA0139.1</td>
<td align="left">19</td>
<td align="left"><a href="https://jaspar2022.genereg.net/matrix/MA0139.1" class="external-link uri">https://jaspar2022.genereg.net/matrix/MA0139.1</a></td>
</tr>
<tr class="odd">
<td align="left">MA1929.1</td>
<td align="left">34</td>
<td align="left"><a href="http://jaspar2022.genereg.net/matrix/MA1929.1" class="external-link uri">http://jaspar2022.genereg.net/matrix/MA1929.1</a></td>
</tr>
<tr class="even">
<td align="left">MA1930.1</td>
<td align="left">35</td>
<td align="left"><a href="http://jaspar2022.genereg.net/matrix/MA1930.1" class="external-link uri">http://jaspar2022.genereg.net/matrix/MA1930.1</a></td>
</tr>
<tr class="odd">
<td align="left">HOCOMOCO v11</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">CTCF_HUMAN.H11MO.0.A</td>
<td align="left">19</td>
<td align="left"><a href="http://hocomoco.autosome.ru/motif/CTCF_HUMAN.H11MO.0.A" class="external-link uri">http://hocomoco.autosome.ru/motif/CTCF_HUMAN.H11MO.0.A</a></td>
</tr>
<tr class="odd">
<td align="left">CTCF_MOUSE.H11MO.0.A</td>
<td align="left">20</td>
<td align="left"><a href="http://hocomoco.autosome.ru/motif/CTCF_MOUSE.H11MO.0.A" class="external-link uri">http://hocomoco.autosome.ru/motif/CTCF_MOUSE.H11MO.0.A</a></td>
</tr>
<tr class="even">
<td align="left">SwissRegulon</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">CTCF.p2</td>
<td align="left">20</td>
<td align="left"><a href="https://swissregulon.unibas.ch/wm/?wm=CTCF.p2&amp;org=hg18" class="external-link uri">https://swissregulon.unibas.ch/wm/?wm=CTCF.p2&amp;org=hg18</a></td>
</tr>
<tr class="even">
<td align="left">Jolma 2013</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">CTCF_full</td>
<td align="left">17</td>
<td align="left"><a href="http://floresta.eead.csic.es/footprintdb/index.php?db=HumanTF:1.0&amp;motif=CTCF_full" class="external-link uri">http://floresta.eead.csic.es/footprintdb/index.php?db=HumanTF:1.0&amp;motif=CTCF_full</a></td>
</tr>
<tr class="even">
<td align="left">CTCFBSDB</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">EMBL_M1, EMBL_M2, MIT_LM2, MIT_LM7, MIT_LM23,
REN_20</td>
<td align="left">9-20</td>
<td align="left"><a href="https://insulatordb.uthsc.edu/download/CTCFBSDB_PWM.mat" class="external-link uri">https://insulatordb.uthsc.edu/download/CTCFBSDB_PWM.mat</a></td>
</tr>
<tr class="even">
<td align="left">CIS-BP</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">83 CTCF (Homo sapiens) C2H2 ZF</td>
<td align="left">11-21</td>
<td align="left"><a href="http://cisbp.ccbr.utoronto.ca/TFreport.php?searchTF=T094831_2.00" class="external-link uri">http://cisbp.ccbr.utoronto.ca/TFreport.php?searchTF=T094831_2.00</a></td>
</tr>
<tr class="even">
<td align="left">2 Ctcf (Mus musculus) C2H2 ZF</td>
<td align="left">15, 20</td>
<td align="left"><a href="http://cisbp.ccbr.utoronto.ca/TFreport.php?searchTF=T100985_2.00" class="external-link uri">http://cisbp.ccbr.utoronto.ca/TFreport.php?searchTF=T100985_2.00</a></td>
</tr>
</tbody>
</table>
<p><strong>CTCF motif logos.</strong> PWMs from (A) MEME, (B) CTCFBSDB,
(C) CIS-BP human, and (D) CIS-BP mouse databases. Clustering and
alignment of motifs was performed using the
<code>r</code>BiocStyle::Biocpkg(“motifStack”)` R package.</p>
<p><img src="../reference/figures/Supplementary_Figure_1.png" width="1126"></p>
<p>See <a href="../inst/scripts/make-data.R">../inst/scripts/make-data.R</a> how
the CTCF GRanges objects were created.</p>
</div>
<div class="section level2">
<h2 id="ctcf-predicted-and-experimental-data">CTCF predicted and experimental data<a class="anchor" aria-label="anchor" href="#ctcf-predicted-and-experimental-data"></a>
</h2>
<p><strong>Predefined CTCF binding data.</strong> “Database” - source of
data; “Number” - number of binding sites; “Assembly” - genome assembly;
“URL” - direct link to data download.</p>
<table class="table">
<colgroup>
<col width="29%">
<col width="5%">
<col width="7%">
<col width="57%">
</colgroup>
<thead><tr class="header">
<th align="left">Database</th>
<th align="right">Number</th>
<th align="left">Assembly</th>
<th align="left">URL</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">CTCFBSDB 2.0</td>
<td align="right">NA</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">Predicted human CTCF binding sites</td>
<td align="right">13401</td>
<td align="left">hg18</td>
<td align="left"><a href="https://insulatordb.uthsc.edu/download/allcomp.txt.gz" class="external-link uri">https://insulatordb.uthsc.edu/download/allcomp.txt.gz</a></td>
</tr>
<tr class="odd">
<td align="left">Predicted mouse CTCF binding sites</td>
<td align="right">5504</td>
<td align="left">mm8</td>
<td align="left"><a href="https://insulatordb.uthsc.edu/download/allcomp.txt.gz" class="external-link uri">https://insulatordb.uthsc.edu/download/allcomp.txt.gz</a></td>
</tr>
<tr class="even">
<td align="left">SCREEN ENCODE</td>
<td align="right">NA</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Human CTCF-bound cCREs</td>
<td align="right">450641</td>
<td align="left">hg38</td>
<td align="left"><a href="https://api.wenglab.org/screen_v13/fdownloads/cCREs/GRCh38-CTCF.bed" class="external-link uri">https://api.wenglab.org/screen_v13/fdownloads/cCREs/GRCh38-CTCF.bed</a></td>
</tr>
<tr class="even">
<td align="left">Mouse CTCF-bound cCREs</td>
<td align="right">82777</td>
<td align="left">mm10</td>
<td align="left"><a href="https://api.wenglab.org/screen_v13/fdownloads/cCREs/mm10-CTCF.bed" class="external-link uri">https://api.wenglab.org/screen_v13/fdownloads/cCREs/mm10-CTCF.bed</a></td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="all-granges-objects-included-in-the-package">All GRanges objects included in the package<a class="anchor" aria-label="anchor" href="#all-granges-objects-included-in-the-package"></a>
</h2>
<p><strong>Summary of CTCF binding data provided in the
package.</strong> CTCF sites for each genome assembly and PWM
combination were detected using FIMO. “ID” - object names formatted as
<code>&lt;assembly&gt;.&lt;database name&gt;</code>; “Assembly” - genome
assembly, T2T - telomere to telomere (GCA_009914755.4) genome assembly;
“All (p-value threshold Xe-Y)” - the total number of CTCF binding sites
in the corresponding BED file at the Xe-Y threshold; “Non-overlapping
(p-value threshold Xe-Y)” - number of non-overlapping CTCF binding sites
(overlapping regions are merged) at the Xe-Y threshold.</p>
<table class="table">
<colgroup>
<col width="16%">
<col width="28%">
<col width="2%">
<col width="3%">
<col width="2%">
<col width="5%">
<col width="8%">
<col width="12%">
<col width="8%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">ID</th>
<th align="left">Description</th>
<th align="left">Genome</th>
<th align="left">Species</th>
<th align="right">Taxonomy</th>
<th align="left">Data.provider</th>
<th align="left">All..p.value.threshold.1e.4.</th>
<th align="left">Non.overlapping..p.value.threshold.1e.4.</th>
<th align="left">All..p.value.threshold.1e.6.</th>
<th align="left">Non.overlapping..p.value.threshold.1e.6.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">T2T.CIS_BP_2.00_Homo_sapiens.RData</td>
<td align="left">T2T CTCF motifs detected using human PWM matrices from
<a href="http://cisbp.ccbr.utoronto.ca/" class="external-link uri">http://cisbp.ccbr.utoronto.ca/</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CIS-BP</td>
<td align="left">85,004,288</td>
<td align="left">11,610,275</td>
<td align="left">1,642,859</td>
<td align="left">308,511</td>
</tr>
<tr class="even">
<td align="left">T2T.CTCFBSDB_PWM.RData</td>
<td align="left">T2T CTCF motifs detected using PWM matrices from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,385,530</td>
<td align="left">3,452,150</td>
<td align="left">98,088</td>
<td align="left">61,030</td>
</tr>
<tr class="odd">
<td align="left">T2T.HOCOMOCOv11_core_HUMAN_mono_meme_format.RData</td>
<td align="left">T2T CTCF motifs detected using human PWM matrices from
<a href="https://hocomoco11.autosome.org/" class="external-link uri">https://hocomoco11.autosome.org/</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">HOCOMOCO v11</td>
<td align="left">927,477</td>
<td align="left">909,247</td>
<td align="left">21,491</td>
<td align="left">21,450</td>
</tr>
<tr class="even">
<td align="left">T2T.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</td>
<td align="left">T2T CTCF motifs detected using human PWM matrices from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">3,196,774</td>
<td align="left">2,538,458</td>
<td align="left">75,126</td>
<td align="left">64,767</td>
</tr>
<tr class="odd">
<td align="left">T2T.Jolma2013.RData</td>
<td align="left">T2T CTCF motifs detected using PWM matrices from <a href="DOI:10.1016/j.cell.2012.12.009" class="uri">DOI:10.1016/j.cell.2012.12.009</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">Jolma 2013</td>
<td align="left">342,765</td>
<td align="left">341,669</td>
<td align="left">6,857</td>
<td align="left">6,857</td>
</tr>
<tr class="even">
<td align="left">T2T.MA0139.1.RData</td>
<td align="left">T2T CTCF motifs detected using MA0139.1 PWM matrix from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">916,829</td>
<td align="left">905,231</td>
<td align="left">22,149</td>
<td align="left">22,131</td>
</tr>
<tr class="odd">
<td align="left">T2T.SwissRegulon_human_and_mouse.RData</td>
<td align="left">T2T CTCF motifs detected using PWM matrices from <a href="https://swissregulon.unibas.ch/sr/" class="external-link uri">https://swissregulon.unibas.ch/sr/</a>, by FIMO</td>
<td align="left">T2T</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">SwissRegulon</td>
<td align="left">1,106,234</td>
<td align="left">1,094,142</td>
<td align="left">23,050</td>
<td align="left">23,006</td>
</tr>
<tr class="even">
<td align="left">hg38.CIS_BP_2.00_Homo_sapiens.RData</td>
<td align="left">hg38 CTCF motifs detected using human PWM matrices from
<a href="http://cisbp.ccbr.utoronto.ca/" class="external-link uri">http://cisbp.ccbr.utoronto.ca/</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CIS-BP</td>
<td align="left">82,804,664</td>
<td align="left">11,502,445</td>
<td align="left">1,612,042</td>
<td align="left">303,446</td>
</tr>
<tr class="odd">
<td align="left">hg38.CTCFBSDB.CTCF_predicted_human.RData</td>
<td align="left">hg38 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,212,674</td>
<td align="left">3,343,301</td>
<td align="left">97,491</td>
<td align="left">60,648</td>
</tr>
<tr class="even">
<td align="left">hg38.CTCFBSDB_PWM.RData</td>
<td align="left">hg38 CTCF motifs detected using PWM matrices from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">13,357</td>
<td align="left">13,278</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="left">hg38.HOCOMOCOv11_core_HUMAN_mono_meme_format.RData</td>
<td align="left">hg38 CTCF motifs detected using human PWM matrices from
<a href="https://hocomoco11.autosome.org/" class="external-link uri">https://hocomoco11.autosome.org/</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">HOCOMOCO v11</td>
<td align="left">891,816</td>
<td align="left">875,038</td>
<td align="left">21,020</td>
<td align="left">20,975</td>
</tr>
<tr class="even">
<td align="left">hg38.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</td>
<td align="left">hg38 CTCF motifs detected using human PWM matrices from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">3,093,041</td>
<td align="left">2,456,234</td>
<td align="left">73,545</td>
<td align="left">63,572</td>
</tr>
<tr class="odd">
<td align="left">hg38.Jolma2013.RData</td>
<td align="left">hg38 CTCF motifs detected using PWM matrices from <a href="DOI:10.1016/j.cell.2012.12.009" class="uri">DOI:10.1016/j.cell.2012.12.009</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">Jolma 2013</td>
<td align="left">330,892</td>
<td align="left">329,834</td>
<td align="left">6,823</td>
<td align="left">6,823</td>
</tr>
<tr class="even">
<td align="left">hg38.MA0139.1.RData</td>
<td align="left">hg38 CTCF motifs detected using MA0139.1 PWM matrix
from <a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">887,980</td>
<td align="left">876,938</td>
<td align="left">21,671</td>
<td align="left">21,652</td>
</tr>
<tr class="odd">
<td align="left">hg38.SCREEN.GRCh38_CTCF.RData</td>
<td align="left">hg38 CTCF-bound human cis-regulatory elements from <a href="https://screen.encodeproject.org/" class="external-link uri">https://screen.encodeproject.org/</a>
</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">ENCODE SCREEN v3</td>
<td align="left">450,641</td>
<td align="left">444,379</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">hg38.SwissRegulon_human_and_mouse.RData</td>
<td align="left">hg38 CTCF motifs detected using PWM matrices from <a href="https://swissregulon.unibas.ch/sr/" class="external-link uri">https://swissregulon.unibas.ch/sr/</a>, by FIMO</td>
<td align="left">hg38</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">SwissRegulon</td>
<td align="left">1,079,055</td>
<td align="left">1,067,050</td>
<td align="left">22,569</td>
<td align="left">22,521</td>
</tr>
<tr class="odd">
<td align="left">hg19.CIS_BP_2.00_Homo_sapiens.RData</td>
<td align="left">hg19 CTCF motifs detected using human PWM matrices from
<a href="http://cisbp.ccbr.utoronto.ca/" class="external-link uri">http://cisbp.ccbr.utoronto.ca/</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CIS-BP</td>
<td align="left">81,569,702</td>
<td align="left">11,355,221</td>
<td align="left">1,595,787</td>
<td align="left">298,489</td>
</tr>
<tr class="even">
<td align="left">hg19.CTCFBSDB.CTCF_predicted_human.RData</td>
<td align="left">hg19 CTCF motifs detected using PWM matrices from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,118,501</td>
<td align="left">3,294,516</td>
<td align="left">93,692</td>
<td align="left">57,137</td>
</tr>
<tr class="odd">
<td align="left">hg19.CTCFBSDB_PWM.RData</td>
<td align="left">hg19 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">13,355</td>
<td align="left">13,294</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">hg19.HOCOMOCOv11_core_HUMAN_mono_meme_format.RData</td>
<td align="left">hg19 CTCF motifs detected using human PWM matrices from
<a href="https://hocomoco11.autosome.org/" class="external-link uri">https://hocomoco11.autosome.org/</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">HOCOMOCO v11</td>
<td align="left">876,822</td>
<td align="left">860,212</td>
<td align="left">20,853</td>
<td align="left">20,811</td>
</tr>
<tr class="odd">
<td align="left">hg19.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</td>
<td align="left">hg19 CTCF motifs detected using human PWM matrices from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">3,035,951</td>
<td align="left">2,414,604</td>
<td align="left">71,825</td>
<td align="left">61,940</td>
</tr>
<tr class="even">
<td align="left">hg19.Jolma2013.RData</td>
<td align="left">hg19 CTCF motifs detected using PWM matrices from <a href="DOI:10.1016/j.cell.2012.12.009" class="uri">DOI:10.1016/j.cell.2012.12.009</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">Jolma 2013</td>
<td align="left">315,914</td>
<td align="left">314,858</td>
<td align="left">6,734</td>
<td align="left">6,734</td>
</tr>
<tr class="odd">
<td align="left">hg19.MA0139.1.RData</td>
<td align="left">hg19 CTCF motifs detected using MA0139.1 PWM matrix
from <a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">871,136</td>
<td align="left">860,252</td>
<td align="left">21,511</td>
<td align="left">21,491</td>
</tr>
<tr class="even">
<td align="left">hg19.SwissRegulon_human_and_mouse.RData</td>
<td align="left">hg19 CTCF motifs detected using PWM matrices from <a href="https://swissregulon.unibas.ch/sr/" class="external-link uri">https://swissregulon.unibas.ch/sr/</a>, by FIMO</td>
<td align="left">hg19</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">SwissRegulon</td>
<td align="left">1,061,972</td>
<td align="left">1,050,085</td>
<td align="left">22,325</td>
<td align="left">22,282</td>
</tr>
<tr class="odd">
<td align="left">hg18.CTCFBSDB.CTCF_predicted_human.RData</td>
<td align="left">hg18 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">hg18</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">13,401</td>
<td align="left">13,343</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">hg18.MA0139.1.RData</td>
<td align="left">hg18 CTCF motifs detected using MA0139.1 PWM matrix
from <a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">hg18</td>
<td align="left">Homo sapiens</td>
<td align="right">9606</td>
<td align="left">JASPAR 2022</td>
<td align="left">869,463</td>
<td align="left">858,603</td>
<td align="left">21,436</td>
<td align="left">21,416</td>
</tr>
<tr class="odd">
<td align="left">mm39.CIS_BP_2.00_Mus_musculus.RData</td>
<td align="left">mm39 CTCF motifs detected using mouse PWM matrices from
<a href="http://cisbp.ccbr.utoronto.ca/" class="external-link uri">http://cisbp.ccbr.utoronto.ca/</a>, by FIMO</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CIS-BP</td>
<td align="left">1,811,582</td>
<td align="left">1,308,209</td>
<td align="left">101,066</td>
<td align="left">80,637</td>
</tr>
<tr class="even">
<td align="left">mm39.CTCFBSDB_PWM.RData</td>
<td align="left">mm39 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,476,211</td>
<td align="left">3,237,788</td>
<td align="left">164,967</td>
<td align="left">104,883</td>
</tr>
<tr class="odd">
<td align="left">mm39.HOCOMOCOv11_core_MOUSE_mono_meme_format.RData</td>
<td align="left">mm39 CTCF motifs detected using mouse PWM matrices from
<a href="https://hocomoco11.autosome.org/" class="external-link uri">https://hocomoco11.autosome.org/</a>, by FIMO</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">HOCOMOCO v11</td>
<td align="left">740,710</td>
<td align="left">726,641</td>
<td align="left">27,620</td>
<td align="left">27,582</td>
</tr>
<tr class="even">
<td align="left">mm39.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</td>
<td align="left">mm39 CTCF motifs detected using mouse PWM matrices from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">JASPAR 2022</td>
<td align="left">2,940,147</td>
<td align="left">2,280,509</td>
<td align="left">109,427</td>
<td align="left">90,585</td>
</tr>
<tr class="odd">
<td align="left">mm39.Jolma2013.RData</td>
<td align="left">mm39 CTCF motifs detected using PWM matrices from <a href="DOI:10.1016/j.cell.2012.12.009" class="uri">DOI:10.1016/j.cell.2012.12.009</a>, by FIMO</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">Jolma 2013</td>
<td align="left">423,521</td>
<td align="left">422,552</td>
<td align="left">12,800</td>
<td align="left">12,797</td>
</tr>
<tr class="even">
<td align="left">mm39.MA0139.1.RData</td>
<td align="left">mm39 CTCF motifs detected using MA0139.1 PWM matrix
from <a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">JASPAR 2022</td>
<td align="left">962,332</td>
<td align="left">949,270</td>
<td align="left">32,210</td>
<td align="left">32,199</td>
</tr>
<tr class="odd">
<td align="left">mm39.SwissRegulon_human_and_mouse.RData</td>
<td align="left">mm39 CTCF motifs detected using PWM matrices from <a href="https://swissregulon.unibas.ch/sr/" class="external-link uri">https://swissregulon.unibas.ch/sr/</a>, by FIMO</td>
<td align="left">mm39</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">SwissRegulon</td>
<td align="left">1,094,643</td>
<td align="left">1,080,229</td>
<td align="left">40,549</td>
<td align="left">40,523</td>
</tr>
<tr class="even">
<td align="left">mm10.CIS_BP_2.00_Mus_musculus.RData</td>
<td align="left">mm10 CTCF motifs detected using mouse PWM matrices from
<a href="http://cisbp.ccbr.utoronto.ca/" class="external-link uri">http://cisbp.ccbr.utoronto.ca/</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CIS-BP</td>
<td align="left">1,810,213</td>
<td align="left">1,307,193</td>
<td align="left">101,011</td>
<td align="left">80,592</td>
</tr>
<tr class="odd">
<td align="left">mm10.CTCFBSDB.CTCF_predicted_mouse.RData</td>
<td align="left">mm10 CTCF motifs detected using PWM matrices from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,472,184</td>
<td align="left">3,235,312</td>
<td align="left">164,855</td>
<td align="left">104,827</td>
</tr>
<tr class="even">
<td align="left">mm10.CTCFBSDB_PWM.RData</td>
<td align="left">mm10 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,502</td>
<td align="left">5,491</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="odd">
<td align="left">mm10.HOCOMOCOv11_core_MOUSE_mono_meme_format.RData</td>
<td align="left">mm10 CTCF motifs detected using mouse PWM matrices from
<a href="https://hocomoco11.autosome.org/" class="external-link uri">https://hocomoco11.autosome.org/</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">HOCOMOCO v11</td>
<td align="left">740,100</td>
<td align="left">726,052</td>
<td align="left">27,604</td>
<td align="left">27,566</td>
</tr>
<tr class="even">
<td align="left">mm10.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</td>
<td align="left">mm10 CTCF motifs detected using mouse PWM matrices from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">JASPAR 2022</td>
<td align="left">2,938,085</td>
<td align="left">2,278,849</td>
<td align="left">109,363</td>
<td align="left">90,532</td>
</tr>
<tr class="odd">
<td align="left">mm10.Jolma2013.RData</td>
<td align="left">mm10 CTCF motifs detected using PWM matrices from <a href="DOI:10.1016/j.cell.2012.12.009" class="uri">DOI:10.1016/j.cell.2012.12.009</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">Jolma 2013</td>
<td align="left">423,194</td>
<td align="left">422,225</td>
<td align="left">12,789</td>
<td align="left">12,786</td>
</tr>
<tr class="even">
<td align="left">mm10.MA0139.1.RData</td>
<td align="left">mm10 CTCF motifs detected using MA0139.1 PWM matrix
from <a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">JASPAR 2022</td>
<td align="left">961,635</td>
<td align="left">948,581</td>
<td align="left">32,188</td>
<td align="left">32,177</td>
</tr>
<tr class="odd">
<td align="left">mm10.SCREEN.mm10_CTCF.RData</td>
<td align="left">mm10 CTCF-bound mouse cis-regulatory elements from <a href="https://screen.encodeproject.org/" class="external-link uri">https://screen.encodeproject.org/</a>
</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">ENCODE SCREEN v3</td>
<td align="left">82,777</td>
<td align="left">82,100</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">mm10.SwissRegulon_human_and_mouse.RData</td>
<td align="left">mm10 CTCF motifs detected using PWM matrices from <a href="https://swissregulon.unibas.ch/sr/" class="external-link uri">https://swissregulon.unibas.ch/sr/</a>, by FIMO</td>
<td align="left">mm10</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">SwissRegulon</td>
<td align="left">1,093,870</td>
<td align="left">1,079,467</td>
<td align="left">40,525</td>
<td align="left">40,499</td>
</tr>
<tr class="odd">
<td align="left">mm9.CIS_BP_2.00_Mus_musculus.RData</td>
<td align="left">mm9 CTCF motifs detected using mouse PWM matrices from
<a href="http://cisbp.ccbr.utoronto.ca/" class="external-link uri">http://cisbp.ccbr.utoronto.ca/</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CIS-BP</td>
<td align="left">1,770,520</td>
<td align="left">1,278,081</td>
<td align="left">99,197</td>
<td align="left">79,218</td>
</tr>
<tr class="even">
<td align="left">mm9.CTCFBSDB.CTCF_predicted_mouse.RData</td>
<td align="left">mm9 CTCF motifs detected using PWM matrices from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,335,475</td>
<td align="left">3,150,407</td>
<td align="left">161,014</td>
<td align="left">102,880</td>
</tr>
<tr class="odd">
<td align="left">mm9.CTCFBSDB_PWM.RData</td>
<td align="left">mm9 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,502</td>
<td align="left">5,491</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
<tr class="even">
<td align="left">mm9.HOCOMOCOv11_core_MOUSE_mono_meme_format.RData</td>
<td align="left">mm9 CTCF motifs detected using mouse PWM matrices from
<a href="https://hocomoco11.autosome.org/" class="external-link uri">https://hocomoco11.autosome.org/</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">HOCOMOCO v11</td>
<td align="left">725,199</td>
<td align="left">711,236</td>
<td align="left">27,103</td>
<td align="left">27,065</td>
</tr>
<tr class="odd">
<td align="left">mm9.JASPAR2022_CORE_vertebrates_non_redundant_v2.RData</td>
<td align="left">mm9 CTCF motifs detected using mouse PWM matrices from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">JASPAR 2022</td>
<td align="left">2,879,784</td>
<td align="left">2,230,170</td>
<td align="left">107,506</td>
<td align="left">89,172</td>
</tr>
<tr class="even">
<td align="left">mm9.Jolma2013.RData</td>
<td align="left">mm9 CTCF motifs detected using PWM matrices from <a href="DOI:10.1016/j.cell.2012.12.009" class="uri">DOI:10.1016/j.cell.2012.12.009</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">Jolma 2013</td>
<td align="left">411,345</td>
<td align="left">410,398</td>
<td align="left">12,309</td>
<td align="left">12,306</td>
</tr>
<tr class="odd">
<td align="left">mm9.MA0139.1.RData</td>
<td align="left">mm9 CTCF motifs detected using MA0139.1 PWM matrix from
<a href="https://jaspar.genereg.net/" class="external-link uri">https://jaspar.genereg.net/</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">JASPAR 2022</td>
<td align="left">938,919</td>
<td align="left">925,939</td>
<td align="left">31,446</td>
<td align="left">31,435</td>
</tr>
<tr class="even">
<td align="left">mm9.SwissRegulon_human_and_mouse.RData</td>
<td align="left">mm9 CTCF motifs detected using PWM matrices from <a href="https://swissregulon.unibas.ch/sr/" class="external-link uri">https://swissregulon.unibas.ch/sr/</a>, by FIMO</td>
<td align="left">mm9</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">SwissRegulon</td>
<td align="left">1,066,676</td>
<td align="left">1,052,374</td>
<td align="left">39,510</td>
<td align="left">39,484</td>
</tr>
<tr class="odd">
<td align="left">mm8.CTCFBSDB.CTCF_predicted_mouse.RData</td>
<td align="left">mm8 CTCF predicted motifs from <a href="https://insulatordb.uthsc.edu/" class="external-link uri">https://insulatordb.uthsc.edu/</a>
</td>
<td align="left">mm8</td>
<td align="left">Mus musculus</td>
<td align="right">10090</td>
<td align="left">CTCFBSDB 2.0</td>
<td align="left">5,504</td>
<td align="left">5,493</td>
<td align="left">-</td>
<td align="left">-</td>
</tr>
</tbody>
</table>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>Below is the citation output from using <code>citation('CTCF')</code>
in R. Please run this yourself to check for any updates on how to cite
<strong>CTCF</strong>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"CTCF"</span><span class="op">)</span>, bibtex <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; To cite package 'CTCF' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Dozmorov MG, Davis E, Mu W, Lee S, Triche T, Phanstiel D, Love M</span></span>
<span><span class="co">#&gt;   (2022). _CTCF_. https://github.com/mdozmorov/CTCF/CTCF - R package</span></span>
<span><span class="co">#&gt;   version 1.00.0, &lt;https://github.com/mdozmorov/CTCF&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {CTCF},</span></span>
<span><span class="co">#&gt;     author = {Mikhail G. Dozmorov and Eric Davis and Wancen Mu and Stuart Lee and Tim Triche and Douglas Phanstiel and Michael Love},</span></span>
<span><span class="co">#&gt;     year = {2022},</span></span>
<span><span class="co">#&gt;     url = {https://github.com/mdozmorov/CTCF},</span></span>
<span><span class="co">#&gt;     note = {https://github.com/mdozmorov/CTCF/CTCF - R package version 1.00.0},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
<!--
Please note that the *[CTCF](https://bioconductor.org/packages/3.16/CTCF)* was only made possible thanks to many other R and 
bioinformatics software authors, which are cited either in the vignettes and/or 
the paper(s) describing this package.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through *[usethis](https://CRAN.R-project.org/package=usethis)*, *[remotes](https://CRAN.R-project.org/package=remotes)*, and *[rcmdcheck](https://CRAN.R-project.org/package=rcmdcheck)* customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and *[BiocCheck](https://bioconductor.org/packages/3.16/BiocCheck)*.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and *[covr](https://CRAN.R-project.org/package=covr)*.
* The [documentation website](http://mdozmorov.github.io/CTCF) is automatically updated thanks to *[pkgdown](https://CRAN.R-project.org/package=pkgdown)*.
* The code is styled automatically thanks to *[styler](https://CRAN.R-project.org/package=styler)*.
* The documentation is formatted thanks to *[devtools](https://CRAN.R-project.org/package=devtools)* and *[roxygen2](https://CRAN.R-project.org/package=roxygen2)*.

For more details, check the `dev` directory.
-->
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.16/biocthis" class="external-link">biocthis</a></em>.</p>
</div>
<div class="section level3">
<h3 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h3>
<p>Please note that the CTCF project is released with a <a href="http://bioconductor.org/about/code-of-conduct/" class="external-link">Contributor Code
of Conduct</a>. By contributing to this project, you agree to abide by
its terms.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] plyranges_1.18.0     GenomicRanges_1.50.0 GenomeInfoDb_1.34.1 </span></span>
<span><span class="co">#&gt;  [4] IRanges_2.32.0       S4Vectors_0.36.0     AnnotationHub_3.6.0 </span></span>
<span><span class="co">#&gt;  [7] BiocFileCache_2.6.0  dbplyr_2.2.1         BiocGenerics_0.44.0 </span></span>
<span><span class="co">#&gt; [10] BiocStyle_2.26.0    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] bitops_1.0-7                  matrixStats_0.62.0           </span></span>
<span><span class="co">#&gt;  [3] fs_1.5.2                      bit64_4.0.5                  </span></span>
<span><span class="co">#&gt;  [5] filelock_1.0.2                httr_1.4.4                   </span></span>
<span><span class="co">#&gt;  [7] rprojroot_2.0.3               tools_4.2.1                  </span></span>
<span><span class="co">#&gt;  [9] bslib_0.4.0                   utf8_1.2.2                   </span></span>
<span><span class="co">#&gt; [11] R6_2.5.1                      DBI_1.1.3                    </span></span>
<span><span class="co">#&gt; [13] withr_2.5.0                   tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt; [15] bit_4.0.4                     curl_4.3.3                   </span></span>
<span><span class="co">#&gt; [17] compiler_4.2.1                textshaping_0.3.6            </span></span>
<span><span class="co">#&gt; [19] cli_3.4.1                     Biobase_2.58.0               </span></span>
<span><span class="co">#&gt; [21] desc_1.4.2                    DelayedArray_0.24.0          </span></span>
<span><span class="co">#&gt; [23] rtracklayer_1.58.0            bookdown_0.29                </span></span>
<span><span class="co">#&gt; [25] sass_0.4.2                    rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt; [27] pkgdown_2.0.6.9000            systemfonts_1.0.4            </span></span>
<span><span class="co">#&gt; [29] stringr_1.4.1                 digest_0.6.30                </span></span>
<span><span class="co">#&gt; [31] Rsamtools_2.14.0              rmarkdown_2.17               </span></span>
<span><span class="co">#&gt; [33] XVector_0.38.0                pkgconfig_2.0.3              </span></span>
<span><span class="co">#&gt; [35] htmltools_0.5.3               MatrixGenerics_1.10.0        </span></span>
<span><span class="co">#&gt; [37] highr_0.9                     fastmap_1.1.0                </span></span>
<span><span class="co">#&gt; [39] rlang_1.0.6                   RSQLite_2.2.18               </span></span>
<span><span class="co">#&gt; [41] shiny_1.7.3                   BiocIO_1.8.0                 </span></span>
<span><span class="co">#&gt; [43] jquerylib_0.1.4               generics_0.1.3               </span></span>
<span><span class="co">#&gt; [45] jsonlite_1.8.3                BiocParallel_1.32.0          </span></span>
<span><span class="co">#&gt; [47] dplyr_1.0.10                  RCurl_1.98-1.9               </span></span>
<span><span class="co">#&gt; [49] magrittr_2.0.3                GenomeInfoDbData_1.2.9       </span></span>
<span><span class="co">#&gt; [51] Matrix_1.5-1                  Rcpp_1.0.9                   </span></span>
<span><span class="co">#&gt; [53] fansi_1.0.3                   lifecycle_1.0.3              </span></span>
<span><span class="co">#&gt; [55] stringi_1.7.8                 yaml_2.3.6                   </span></span>
<span><span class="co">#&gt; [57] SummarizedExperiment_1.28.0   zlibbioc_1.44.0              </span></span>
<span><span class="co">#&gt; [59] grid_4.2.1                    blob_1.2.3                   </span></span>
<span><span class="co">#&gt; [61] parallel_4.2.1                promises_1.2.0.1             </span></span>
<span><span class="co">#&gt; [63] crayon_1.5.2                  lattice_0.20-45              </span></span>
<span><span class="co">#&gt; [65] Biostrings_2.66.0             KEGGREST_1.38.0              </span></span>
<span><span class="co">#&gt; [67] knitr_1.40                    CTCF_1.00.0                  </span></span>
<span><span class="co">#&gt; [69] pillar_1.8.1                  rjson_0.2.21                 </span></span>
<span><span class="co">#&gt; [71] codetools_0.2-18              XML_3.99-0.12                </span></span>
<span><span class="co">#&gt; [73] glue_1.6.2                    BiocVersion_3.16.0           </span></span>
<span><span class="co">#&gt; [75] evaluate_0.17                 BiocManager_1.30.19          </span></span>
<span><span class="co">#&gt; [77] png_0.1-7                     vctrs_0.5.0                  </span></span>
<span><span class="co">#&gt; [79] httpuv_1.6.6                  purrr_0.3.5                  </span></span>
<span><span class="co">#&gt; [81] assertthat_0.2.1              cachem_1.0.6                 </span></span>
<span><span class="co">#&gt; [83] xfun_0.34                     mime_0.12                    </span></span>
<span><span class="co">#&gt; [85] xtable_1.8-4                  restfulr_0.0.15              </span></span>
<span><span class="co">#&gt; [87] later_1.3.0                   ragg_1.2.4                   </span></span>
<span><span class="co">#&gt; [89] tibble_3.1.8                  GenomicAlignments_1.34.0     </span></span>
<span><span class="co">#&gt; [91] AnnotationDbi_1.60.0          memoise_2.0.1                </span></span>
<span><span class="co">#&gt; [93] ellipsis_0.3.2                interactiveDisplayBase_1.36.0</span></span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Mikhail Dozmorov, Eric Davis, Wancen Mu, Stuart Lee, Tim Triche, Douglas Phanstiel, Michael Love.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.9000.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
